lib/pool.js

```diff
@@ -27,21 +27,19 @@ function _addBackoff(factory) {
 
   factory.create = function __run() {
 
-    return originalCreateMethod().catch((e) =>
+    return originalCreateMethod().catch(function ___handleCreateError(e) {
 
-      new Bluebird(function ___backoffRunResolver(resolve, reject) {
-        const wait = backoff.next()
+      const wait = backoff.next()
```

Comment:
I just noticed that backoff is created each time. The `next()` called on the newly created instance would return the same value wouldn't it?